File: ska_frontend/src/gui/page_assistant/dtos/dto_ask_assistant_question.rs
use serde::{Deserialize, Serialize};

#[derive(Clone, Serialize, Deserialize)]
pub struct AskAssistantQuestionRequest {
    pub question: String,
    pub llm_model: String,
    pub prompt_template: Option<String>,
}

#[derive(Clone, Serialize, Deserialize)]
pub struct AskAssistantQuestionResponse {
    pub answer: String,
}

File: ska_frontend/src/gui/page_assistant/dtos/dto_ask_assistant_question.ts
import { UtilsTypes } from "@/utils/core/utils_types";
import { DtoChatPacket } from "./dto_chat_packet";

export class AskAssistantQuestionRequest {
    chat_id: number;
    question: string;

    constructor(obj: {
        chat_id: number,
        question: string,
    }) {
        this.chat_id = obj.chat_id;
        this.question = obj.question;
    }
}

export class AskAssistantQuestionResponse {
    question: DtoChatPacket;
    answer: DtoChatPacket;

    constructor(obj: {
        question: DtoChatPacket;
        answer: DtoChatPacket;
    }) {
        this.question = obj.question;
        this.answer = obj.answer;
    }

    static fromUnknown(value: unknown): AskAssistantQuestionResponse {
        const obj = UtilsTypes.unknownToObject(value).unwrap();
        return new AskAssistantQuestionResponse({
            question: DtoChatPacket.fromUnknown(obj.question),
            answer: DtoChatPacket.fromUnknown(obj.answer)
        })
    }
}
